assets/src/models/createDetourMachine.ts

```diff
@@ -1,23 +1,114 @@
-import { setup } from "xstate"
+import { setup, assign } from "xstate"
+import { ShapePoint } from "../schedule"
 
 export const createDetourMachine = setup({
   types: {} as {
-    events: { type: "detour.share" } | { type: "detour.resume-editing" }
+    context: {
+      waypoints: ShapePoint[]
+    }
+    events:
+      | { type: "detour.share" }
+      | { type: "detour.resume-editing" }
+      | { type: "detour.edit.place-waypoint"; location: ShapePoint }
+      | { type: "detour.edit.place-connection-point"; location: ShapePoint }
+      | { type: "detour.edit.clear-detour" }
+      | { type: "detour.edit.undo" }
+      | { type: "detour.edit.done" }
+      | { type: "detour.share.copy-detour"; detourText: string }
```

Comment:
tiny: would it make sense to alphabetize this list so that the two `share` events are co-located?

Reply:
I intentionally sorted them this way, and but I'd like to reconsider some of these event names, so if you have any suggestions!

Reply:
Turns out, I don't actually use `detour.share`! So we can delete it altogether!

Reply:
out of curiosity, what was the sorting?

Reply:
Upon reflection, It might have been "Just Vibes™️" of what felt right, I'm rethinking this and am preparing to sort these alphabetically, thanks for pointing this out!

Reply:
Sorted!